### PR TITLE
fix(cli/upgrade): properly cleanup after finished

### LINKED ISSUE #15342: `deno upgrade` should properly cleanup
Currently, `deno upgrade` doesn't remove the downloaded zip (and uncompressed Deno binary if `--dry-run` is specified) when finished. This is bad practice and occupies space:

- On macOS, temporary directory is on disk and only cleaned on next boot;
- On Windows, temporary directory is on disk and (I believe) only occasionally cleaned;
- On Linux, temporary directory is (usually) in memory

### PR DESCRIPTION
**Pull request by kidonng**
Fix #15342

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   },
 
   "extensions": [
-    "matklad.rust-analyzer",
+    "rust-lang.rust-analyzer",
```
Comment (kidonng) [inline .devcontainer/devcontainer.json:17]: Unrelated, but I noticed that rust-analyzer isn't installed in the devcontainer, so I updated the extension id (see https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer).

Comment (bartlomieju) [reply .devcontainer/devcontainer.json:17]: Since it's unrelated to this PR I suggest to open a new PR that will change it

Comment (kidonng) [reply .devcontainer/devcontainer.json:17]: It's just one line though? Doesn't seem to worth the hassle.

Comment (bartlomieju) [reply .devcontainer/devcontainer.json:17]: If someone complains that it broke their setup we will revert a commit including changes to "upgrade" subcommand. So yes, it's worth the hassle

Comment (bartlomieju) [reply .devcontainer/devcontainer.json]: Addressed in https://github.com/denoland/deno/pull/15751

**Pull request closed without merging**